File: C++/Medium/Accepted/eliminate-maximum-number-of-monsters_9.cpp
class Solution {
public:
    int eliminateMaximum(vector<int>& dist, vector<int>& speed) {
        
        int n = dist.size();
        vector<double> time(n);
        for(int i=0; i<n; i++) {
            time[i] = (double)dist[i]/speed[i];
        }
        sort(time.begin(), time.end());
        int i = 0;
        for(; i<n; i++) {
            if(time[i] <= i) {
                break;
            }
        }
        return i;
    }
};

File: Java/Medium/Accepted/eliminate-maximum-number-of-monsters_9.java
class Solution {
    public int eliminateMaximum(int[] dist, int[] speed) {
        
        int n=dist.length;
        int[] time=new int[n];
        for(int i=0;i<n;i++){
            time[i]=dist[i]/speed[i];
            if(dist[i]%speed[i]!=0)
                time[i]++;
        }
        Arrays.sort(time);
        int cnt=0;
        for(int i=0;i<n;i++){
            if(time[i]<=i)
                break;
            cnt++;
        }
        return cnt;
    }
}